renku/cli/init.py

```diff
@@ -64,128 +64,255 @@
 
 
+def parse_variables(ctx, param, value):
+    """Parse template variables to dictionary."""
+    try:
+        variables = ast.literal_eval(value)
+    except ValueError:
+        raise_template_error(value)
+    if type(variables) is not dict:
+        raise_template_error(value)
+    return variables
+
+
+def raise_template_error(value):
+    """Raise template error with short explanation."""
+    error_info = [
+        '{0}'.format(value), 'Tip: a dictionary is expected',
+        (
+            'Example: --template-variables '
+            '\'{ "variable_1": "string", "variable_2": 2 }\''
+        )
+    ]
+    raise errors.ParameterError(
+        '\n'.join(error_info), '"--template-variables"'
+    )
+
+
 def validate_name(ctx, param, value):
     """Validate a project name."""
     if not value:
-        value = os.path.basename(ctx.params['directory'].rstrip(os.path.sep))
+        value = os.path.basename(ctx.params['path'].rstrip(os.path.sep))
     return value
 
 
-def store_directory(ctx, param, value):
+def store_directory(value):
     """Store directory as a new Git home."""
     Path(value).mkdir(parents=True, exist_ok=True)
     set_git_home(value)
     return value
 
 
-def template(client, force):
-    """Render templated configuration files."""
-    import pkg_resources
+def create_template_sentence(templates, instructions=False):
+    """Create templates choice sentence.
+
+    :ref templates: list of templates coming from manifest file
+    :ref instructions: add instructions
+    """
+    Template = namedtuple('Template', ['index', 'name', 'description'])
+    templates_friendly = [
+        Template(
+            index=index + 1,
+            name=template_elem['name'],
+            description=template_elem['description'],
+        ) for index, template_elem in enumerate(templates)
+    ]
+
+    text = tabulate(
+        templates_friendly,
+        headers=OrderedDict((
+            ('index', 'Number'),
+            ('name', 'Name'),
+            ('description', 'Description'),
+        ))
+    )
 
-    # create the templated files
-    for tpl_file in CI_TEMPLATES:
-        tpl_path = client.path / tpl_file
-        with pkg_resources.resource_stream(__name__, tpl_file) as tpl:
-            content = tpl.read()
+    if not instructions:
+        return text
+    return '{0}\nPlease choose a template by typing the number'.format(text)
 
-            if not force and tpl_path.exists():
-                click.confirm(
-                    'Do you want to override "{tpl_file}"'.format(
-                        tpl_file=tpl_file
-                    ),
-                    abort=True,
-                )
 
-            with tpl_path.open('wb') as dest:
-                dest.write(content)
+def is_path_empty(path):
+    """Check if path contains files.
+
+    :ref path: target path
+    """
+    gen = Path(path).glob('**/*')
+    try:
+        next(gen)
+        return False
+    except StopIteration:
+        return True
```

Comment:
```
return not any(gen)
```

Would accomplish the same without having to deal with exceptions

Reply:
Thanks for the cleanup suggestion!

Reply:
does `any(gen)` create a list first though?

Reply:
No, it calls iternext(...) and checks if that's NULL or not (in cython C code). So basically it does the same, just without having to deal with exceptions.

Any returns true on the first element encountered that evaluates to true, so if paths was `['', '']` there'd be a difference (above code returning False and `not any(...)` returning True, but I don't think that matters here.

And I find it more readable with `any`